### PR TITLE
tests: test for gate-auto-refresh hook error resulting in hold

### PR DESCRIPTION
**Pull request by stolowski**
Test the case where gate-auto-refresh hook errors out, resulting in hold, and for case where the hook does nothing (resulting in normal refresh). It adds a 3rd version of the test snap to test new scenarios (the snap has been uploaded to the store).

### CODE REVIEW

File: tests/main/auto-refresh-gating/task.yaml
Comment (mardy) [inline tests/main/auto-refresh-gating/task.yaml:62]: nitpick: you could use the variables to get the snap names here too

File: tests/main/auto-refresh-gating/task.yaml
Comment (mardy) [inline tests/main/auto-refresh-gating/task.yaml:79]: Since this is the first fime that you use `$CONTROL_FILE` in this test, it might be a good diea to add a comment to let the reader know what this does. Something like:

```
# Writing into this file affects the command performed by the gate-auto-refresh hook
# in tests/lib/snaps/store/test-snapd-refresh-control.v*/meta/hooks/gate-auto-refresh
```

**Review by mardy — APPROVED** [review]
Looks good, just minor comments.

**Comment by stolowski** [comment]
Blocked until the prerequisite lands.

File: tests/main/auto-refresh-gating/task.yaml
Comment (stolowski) [reply tests/main/auto-refresh-gating/task.yaml:34]: No, it shouldn't I think (thanks to snap debug ensure-state-soon). I've reduced it to 200 in the prerequisite PR that already landed.

File: tests/main/auto-refresh-gating/task.yaml
Comment (stolowski) [reply tests/main/auto-refresh-gating/task.yaml:91]: Added

File: tests/main/auto-refresh-gating/task.yaml
Comment (stolowski) [reply tests/main/auto-refresh-gating/task.yaml]: Indeed, done in the prereq PR.

File: tests/main/auto-refresh-gating/task.yaml
Comment (sergiocazzolato) [inline tests/main/auto-refresh-gating/task.yaml:207]: LAST_REFRESH_CHANGE_ID is not used after this, should you check that it is different from the previous one?

File: tests/main/auto-refresh-gating/task.yaml
Comment (sergiocazzolato) [inline tests/main/auto-refresh-gating/task.yaml:194]: do you need to check this twice? both seem to be pretty similar

File: tests/main/auto-refresh-gating/task.yaml
Comment (stolowski) [reply tests/main/auto-refresh-gating/task.yaml:207]: That is already checked inside wait_for_autorefresh.

File: tests/main/auto-refresh-gating/task.yaml
Comment (stolowski) [reply tests/main/auto-refresh-gating/task.yaml:194]: Yep, I've just fixed this. Thanks

File: tests/main/auto-refresh-gating/task.yaml
Comment (sergiocazzolato) [reply tests/main/auto-refresh-gating/task.yaml:207]: ah, nice

**Review by sergiocazzolato — APPROVED** [review]
Thanks

**Pull request merged**
